eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/tools/autocomplete/AutocompleteSupervisor.java

```diff
@@ -0,0 +1,41 @@
+package ch.cern.cmms.eamlightweb.tools.autocomplete;
+
+import ch.cern.cmms.eamlightweb.tools.EAMLightController;
+import ch.cern.cmms.eamlightweb.tools.interceptors.RESTLoggingInterceptor;
+import ch.cern.eam.wshub.core.services.grids.entities.GridRequest;
+import ch.cern.eam.wshub.core.services.grids.entities.GridRequestFilter;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.interceptor.Interceptors;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.Arrays;
+
+@Path("/autocomplete")
+@ApplicationScoped
+@Interceptors({RESTLoggingInterceptor.class})
+public class AutocompleteSupervisor extends EAMLightController {
+
+    @GET
+    @Path("/supervisor/{code}")
+    @Produces("application/json")
+    @Consumes("application/json")
+    public Response complete(@PathParam("code") String code) {
+        GridRequest gridRequest = new GridRequest("LVSCHEDGROUP", GridRequest.GRIDTYPE.LOV, 10);
+        String uppercasedCode = code.toUpperCase();
+
+        gridRequest.addFilter("schedgroup", uppercasedCode, "BEGINS", GridRequestFilter.JOINER.OR);
+
+        Arrays.stream(uppercasedCode.split(" ")).forEach(name -> {
+            gridRequest.addFilter("schedgroupdesc", " " + name, "CONTAINS",
+                    GridRequestFilter.JOINER.OR, true, false);
+
+            gridRequest.addFilter("schedgroupdesc", name, "BEGINS",
+                    GridRequestFilter.JOINER.AND, false, true);
+        });
+
+        gridRequest.sortBy("schedgroupdesc");
```

Comment:
Is the performance impact by adding the sorting worth it for the added value?

Reply:
I've done some testing with Postman and I've got the following results.

Local Performance Test (100 iterations and 0ms between iteration):
- Without filtering and **with sorting**: 204ms
- Without filtering and **without sorting**: 204ms

The results show that there is no difference between sorting them or not at least with the number of supervisors currently in the system.